src/provider/rpc.ts

```diff
@@ -231,14 +232,37 @@ export class RpcProvider implements ProviderInterface {
       throw Error('getContractVersion require contractAddress or classHash');
     }
 
+    // Take the opportunity of class reading, to get the name of the message signature verification name
+    // or an empty string in case of proxy.
+    let messageVerifFunctionName: string | undefined;
+    if ('signatureVerifFunctionName' in this) {
+      const parser = createAbiParser(contractClass.abi);
```

Comment:
Looks like there is enough overlap here with the code added in the Account `verifyMessageHash` method that could be extracted to a utility method, it would do a superfluous proxy check for that method but it shouldn't be too costly.

Reply:
Are you sure?
In detail, seems rather different...